src/framework/anim/evaluator/anim-clip.js

```diff
@@ -66,6 +66,7 @@ class AnimClip {
 
     set speed(speed) {
         this._speed = speed;
+        this.alignCursorToCurrentTime();
```

Comment:
should this be called only when the sign changes?

Reply:
but even then, it seems strange to change the start index for events when the speed changes, as at this point the time has not changed?

Reply:
The clip events are evaluated using an incoming delta time and the current speed. The clip's time will not necessarily have changed when this evaluation happens. That'll only occur if the user has actively set the time themselves.